### PR TITLE
Upgrade to Kotlin 2.0.0

### PR DESCRIPTION
**Pull request by sgrimm**
Build the code with the new, faster, Kotlin compiler.

No changes to our application code were needed, but by default, jOOQ generates
Kotlin code with annotations that are flagged as warnings by the new compiler.
Turn off the code generator option that enables the annotations; they are only
for bidirectional Java interop and aren't needed for pure-Kotlin projects.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#1980** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1980?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by sgrimm** [comment]
Won't merge this until after the next release so it has more time to bake in staging.

File: build.gradle.kts
Comment (sgrimm) [inline build.gradle.kts:222]: https://github.com/jOOQ/jOOQ/issues/16700

**Review by nickgraz — APPROVED** [review]

**Pull request merged**
